src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -644,7 +644,7 @@ private long implTransferTo(OutputStream out) throws IOException {
             int avail = count - pos;
             if (avail > 0) {
                 if (isTrusted(out)) {
-                    out.write(getBufIfOpen(), pos, count);
+                    out.write(getBufIfOpen(), pos, avail);
                 } else {
                     // Prevent poisoning and leaking of buf
                     byte[] buffer = Arrays.copyOfRange(getBufIfOpen(), pos, count);
```

Comment:
@bplb Shouldn't it be `avail` *here*, too?

Reply:
No: the third param of [Arrays.copyOfRange](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Arrays.html#copyOfRange(byte[],int,int)) is `to`, not `len`.

Reply:
Ah, this explains why it did not fail originally, but only after adding the "isTrusted" case! 🙏

Reply:
See https://github.com/openjdk/jdk/pull/17250#issuecomment-1875761630 above.